common/helpers/g2p.py

```diff
@@ -0,0 +1,102 @@
+__all__ = ["G2PBasedOovAugmenter"]
+
+import os
+from typing import Optional
+
+from sisyphus import tk
+Path = tk.setup_path(__package__)
+
+from i6_core.corpus.stats import ExtractOovWordsFromCorpusJob
+from i6_core.g2p.apply import ApplyG2PModelJob
+from i6_core.g2p.convert import BlissLexiconToG2PLexiconJob, G2POutputToBlissLexiconJob
+from i6_core.g2p.train import TrainG2PModelJob
+
+
+class G2PBasedOovAugmenter():
+    def __init__(self,
+                 original_bliss_lexicon,
+                 g2p_model_path:Optional[str] = None,
+                 train_args:Optional[dict] = None,
+                 apply_args:Optional[dict] = None,
+                 ):
+        """
+        :param original_bliss_lexicon: path to the original lexicon with OOV
+        :param g2p_model_path: path to the g2p model, if none a g2p model is trained
+        #######################################
+        :param train_args = {
+        "num_ramp_ups"   :4,
+        "min_iter"       :1,
+        "max_iter"       :60,
+        "devel"          :"5%",
+        "size_constrains":"0,1,0,1"
+        }
+        #######################################
+        :param apply_args = {
+        "variants_mass"  :1.0,
+        "variants_number":1
+        }
+        """
+        super().__init__()
+        self.original_bliss_lexicon = original_bliss_lexicon
+        self.g2p_model_path = g2p_model_path
+        self.train_args = {}
+        self.apply_args = {}
+
+        if train_args is not None:
+            self.train_args.update(train_args)
+
+        if apply_args is not None:
+            self.apply_args.update(apply_args)
+
+
+    def train_and_set_g2p_model(self, train_lexicon:str, alias_path:str):
+        g2p_lexicon_job = BlissLexiconToG2PLexiconJob(bliss_lexicon=train_lexicon)
+
+        g2p_train_job = TrainG2PModelJob(
+            g2p_lexicon=g2p_lexicon_job.out_g2p_lexicon,
+            **self.train_args)
+        g2p_train_job.add_alias(os.path.join(alias_path, "train_g2p_model"))
+        self.g2p_model_path = g2p_train_job.out_best_model
+
+
+    def get_g2p_augmented_bliss_lexicon(
+            self,
+            bliss_corpus:Path,
+            corpus_name: str,
+            alias_path:str,
+            train_lexicon:Optional[str] = None,
```

Comment:
yes i would agree with this.

Reply:
I do not understand your comment. As you can see the class does not have separate attributes for `train_lexicon` and `original_bliss_lexicon` (or how @michelwi call seed_lexicon). The idea is that you instantiate the class for the lexicon with respect to which you want to identify your OOVs, when you call this function you can decide whether your model is trained with the same lexicon or another lexicon.

Reply:
> you can decide whether your model is trained with the same lexicon or another lexicon.
but this is not consistent. the lexicon used to get the OOVs should be the same used to train the g2p. I can not think of a case where you need to use another lexicon.

Reply:
> lexicon used to get the OOVs should be the same used to train the g2p

Here, the idea is that you might want to use an already existing sis job, and instead of providing the path of the model, it is cleaner to use just the input lexicon and let the rest to the class. 
The option of providing the model path is for cases we use external pretrained models

Reply:
> I can not think of a case where you need to use another lexicon

As @michelwi pointed out, you might have this case.

Reply:
but in @michelwi he still uses the `seed_lex` to get the OOVs so it is still consistent. In your case, you use `self.original_bliss_lexicon`.

Reply:
But then this is exactly the wrong way. The class should get the "training" lexicon, and this function can then get an  "reference" (or seed, although I do not get the term) lexicon for the OOVs that is then extended and returned

Reply:
> The class should get the "training" lexicon

If you see how the class is designed and named, you can see that it is not g2p-model-centered but lexicon-centered. This is why the attribute of the class from my point of view should not be the train lexicon but the lexicon you want to augment.

Reply:
The class name definitely does not say this in my opinion... So in my head it is much more logical to have it g2p-model centered. Because otherwise the train and apply args would not make much sense.

Reply:
> So in my head it is much more logical to have it g2p-model

Ok. Let's do it in this way.

Reply:
"seed" because it grows when you add more words to it.

Also note, that in my code in each iteration the seed_lexicon is overwritten with the updated lexicon. (See sorting of lemmata, s.t. we can reuse alignments made with the smaller corpus after more and more oov)

@Marvin84 I see that your code is centered around the lexicon-to-be-augmented. What I (we) are arguing that it should rather be around the g2p model, because in practice (possibility only my practice) I want to have only one model but continuously add (append) more words to it.

Reply:
> What I (we) are arguing that it should rather be around the g2p model

Yes, but then that should be implemented differently. I am adapting this implementation to your idea. The idea is that there is an additional optional parameter for `train_lexicon`. In case it is not set then this is the same as the `original_bliss_lexicon`. Then if the model path is None, a model is trained.